src/signed.rs

```diff
@@ -192,54 +247,86 @@ impl<T: Signable> Index for Indexed<T> {
 }
 
 #[derive(Debug)]
-pub struct PartiallyMultisigned<'a, T: Signable, MK: MultiKeychain> {
-    unchecked: UncheckedSigned<T, MK::PartialMultisignature>,
-    marker: PhantomData<&'a MK>,
-}
-
 pub struct Multisigned<'a, T: Signable, MK: MultiKeychain> {
     pub unchecked: UncheckedSigned<T, MK::PartialMultisignature>,
     marker: PhantomData<&'a MK>,
 }
 
+impl<'a, T: Signable + Clone, MK: MultiKeychain> Clone for Multisigned<'a, T, MK> {
+    fn clone(&self) -> Self {
+        Multisigned {
+            unchecked: self.unchecked.clone(),
+            marker: self.marker,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct IncompleteMultisignatureError<'a, T: Signable, MK: MultiKeychain> {
+    pub(crate) partial: PartiallyMultisigned<'a, T, MK>,
+}
+
 #[derive(Debug)]
-pub struct IncompleteMultisignatureError<'a, T: Signable, MK: MultiKeychain> {
-    pub partial: PartiallyMultisigned<'a, T, MK>,
+pub enum PartiallyMultisigned<'a, T: Signable, MK: MultiKeychain> {
```

Comment:
Disagree. Typically "partial" does not exclude possibility of completeness (like in partial function or partial order)

Reply:
Feels a bit strange to me nonetheless (I think because in your examples you usually use only the properties implied by "partialness", while here you explicitly have "complete" as one of the options), but I'm not gonna insist.